File: src/IsbInterop/References/Proxies/CompRecordInfo.vb
Imports IsbInterop.Base.Proxies

Namespace References.Proxies

    ''' <summary>
    ''' Обертка над ICompRecordInfo.
    ''' </summary>
    Friend MustInherit Class CompRecordInfo
        Inherits ObjectInfo
        Implements ICompRecordInfo

        ''' <summary>
        ''' Код записи.
        ''' </summary>
        Public ReadOnly Property Code As String Implements ICompRecordInfo.Code
            Get
                Return DirectCast(GetRcwProperty("Code"), String)
            End Get
        End Property

        ''' <summary>
        ''' Конструктор.
        ''' </summary>
        ''' <param name="rcwICompRecordInfo">COM-объект ICompRecordInfo.</param>
        ''' <param name="scope">Область видимости.</param>
        Protected Sub New(rcwICompRecordInfo As Object, scope As IScope)
            MyBase.New(rcwICompRecordInfo, scope)
        End Sub
    End Class
End Namespace

File: src/IsbInterop/References/Proxies/CompRecordInfo.cs
using IsbInterop.Base.Proxies;

namespace IsbInterop.References.Proxies
{
  /// <summary>
  /// Обертка над ICompRecordInfo.
  /// </summary>
  internal abstract class CompRecordInfo : ObjectInfo, ICompRecordInfo
  {
    /// <summary>
    /// Код записи.
    /// </summary>
    public string Code => (string)GetRcwProperty("Code");

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwICompRecordInfo">COM-объект ICompRecordInfo.</param>
    /// <param name="scope">Область видимости.</param>
    protected CompRecordInfo(object rcwICompRecordInfo, IScope scope)
      : base(rcwICompRecordInfo, scope) { }
  }
}
